src/kern/exec.c

```diff
@@ -53,6 +68,32 @@ int BPF_PROG(test_void_hook, struct linux_binprm *bprm)
 	// Increment the previous value, as a new process was executed on the
 	// current CPU.
 	(*val)++;
+
+	// Get information about the current process
+	int pid = bpf_get_current_pid_tgid() >> 32;
+	int tgid = (bpf_get_current_pid_tgid() << 32) >> 32;
+	int ppid = 1;
+	struct record_sample *sample;
+
+	sample = bpf_ringbuf_reserve(&ringbuf, sizeof(*sample), flags);
```

Comment:
This should not return an integer. Also, no need for the brackets here.

Reply:
Can you give me more information about these?

bpf_ringbuf_reserve returns a pointer to the start of the memory with size bytes of memory available, it does not return an integer.

Also, what brackets? I used brackets for functions, they are needed.

Reply:
The LSM hook bprm_committed_creds does not return a value, where as your BPF program returns an integer. So you need to:

- change the return type of the program
- not return an int in this  in this error block

So:
  
if (!sample) 
    return;

RE Brackets you don't need brackets in the above statement. It's a minor style thing.